### ISSUE TITLE
Syringe not working.

### ISSUE THREAD

**Issue by unpure** [McJty/RFTools#245]
Hey there.
I'm currently trying to use the environmental controller in my base.
Sadly it seems like i cant use the syringe on any mob.
I found some issues about people that have problems with the syringe and wrong recipes.
But atleast they can use the syringe. I can try everything but it does not fill with essence.
I'm using the latest version of RFtools and forge 10.13.4.1481.
A full list of mods that are used in my modpack can be found here:
http://upclan.de/modpack/info/
Maybe its an issue with any of that mods?

Sorry for my bad english, i hope its not to hard to understand.

**Comment by McJty** [comment]
How are you using it? You need to hit the mob with it (left click) and not right click

**Comment by unpure** [comment]
Yeah i know that. It was working a few versions back, but at some point it stopped and i dont know why.

**Comment by unpure** [comment]
Any news about the syringe problem?

**Comment by McJty** [comment]
I cannot reproduce this. Works perfectly fine for me. So no idea really. On what mob are you trying to use it?

**Comment by unpure** [comment]
I tried it on zombies, skeletons, creeper, sheep and iron golem.
If its not a but with that mobs, it seems to be not working at all.

**Comment by McJty** [comment]
I just tried it on a cow in my dev environment and it works perfectly. Also I've had no other reports on this. So strange. No clue right now...

**Comment by unpure** [comment]
Then it has to be a mod incompatibility i think.
Will try to find the mod that causes the problem later.

**Comment by unpure** [comment]
So after a lot of testing, this problem is NOT a mod incompatibility.
I decided to remove a few dimlets with the knowndimlets config option in my modpack.
I love to create new dimensions to play in, but my modpack needs a lot of balance to work.
Because of that i removed all dimlets that are not needed to create a normal dimension, like most of the liquids/materials and many other dimlets.
It seems that the syringe gets bugged if the mob dimlets are removed.
Adding
´´´
I:dimlet.Mob.Zombie=-1
´´´
to the knowndimlets option, creates the bug with syringes for zombies.
Obviously every other mob dimlets that gets added creates the same bug for every other mog.
So after removing zombies and iron golems from the knowndimlets option, lets me finally use the syringe at these mobs. I dont know if this is intended to work this way, but i dont think/hope so.

tl;dr
restricting mob dimlets ends in useless syringes.

**Comment by McJty** [comment]
Ah indeed. Syringes only work for mobs for which there is a dimlet. That's hard to avoid. However, there are some things you can do. You can add the following to dimlets.cfg:

B:"expensive.Mob.Skeleton"=true

That will disable the dimlet from appearing randomly in dimensions.

In the next version of RFTools you can also do this:

B:"noloot.Mob.Skeleton"=true

That way the dimlet will never be generated randomly (researcher, dimlet house, ...)

Is this an adequate solution then?

**Comment by unpure** [comment]
So where does this "expensive" entry needs to be placed? dimletsettings or knowndimlets?
Is there any need for the expensive entry in the next version or does it needs to be replaced with the noloot entry then? Just so i know that i have to change it then.

To understand it correct, the expensive option only removes it from the dimlet house, but can still be researched? And in the next version the noloot will stop it from beeing generated in a dimlet house and from the researcher. But is there any other way to get it, or is it completly disabled without cheats? If it is completly disabled, that would be a good solution.

**Comment by McJty** [comment]
No, the expensive option removes it from worldgen. That means the mob feature will not be selected in a random dimension. It doesn't remove the dimlet from houses. That's what the 'noloot' will do.

And you add it to the 'dimletsettings' section.

With those two options there would be no other way to get it except with the dimlet workbench but if you disable all mob dimlets like that then you will not even be able to do that because you need to extract other mob dimlets to make new mob dimlets.

**Comment by unpure** [comment]
So in the next version i'm going to set every mob dimlet on expensive and noloot to true, this will "disable" all mob dimlets but lets me create the syringe and use the syringes to set the spawner or create modules. Sounds good to me, thats all i want! :)

Thank you very much!
